### PR TITLE
Fix test framework, add more test graphs for new engine

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

- Add more test graphs to test new engine
- Add support for inspecting node values for multi-outport nodes with new engine
- Throw assertion exceptions from new, refactored `SerializationUtils.CompareWorkspaceMethod`, without which all tests were always passing irrespective of the results

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: test/DynamoCoreTests/SerializationTests.cs [hunk 1]
```diff
@@ -186,12 +186,27 @@ private void Init(WorkspaceModel workspace, EngineController controller, bool ds
...
                         {
-                            // TODO: Add support for multi-outport nodes.
-                           //foreach(var p in n.OutPorts)
-                           {
-                                var objs = n.GetCSValue(/*p.Index*/0, controller).Cast<object>();
-                                portvalues.AddRange(objs.Select(x => x.ToString()));
-                           };
+                            portvalues.AddRange(n.OutPorts.Select(p =>
+                            {
+                                var objs = n.GetCSValue(p.Index, controller).Cast<object>();
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/SerializationTests.cs:191]: what does `CS`? value mean?

File: test/DynamoCoreTests/SerializationTests.cs [hunk 2]
```diff
@@ -323,6 +334,8 @@ public static void CompareWorkspaceModels(serializationTestUtils.WorkspaceCompar
                 }
                 catch
                 {
+                    if (!dsExecution) throw;
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/SerializationTests.cs:337]: should this re throw the caught exception?

**Review by mjkkirschner — APPROVED** [review]

File: test/DynamoCoreTests/SerializationTests.cs (hunk 1 above)
Comment (aparajit-pratap) [reply test/DynamoCoreTests/SerializationTests.cs:191]: CSharp for want of a better name, I'm open to changing it.

Comment (aparajit-pratap) [reply test/DynamoCoreTests/SerializationTests.cs:191]: I couldn't use GetValue as it conflicts with an existing method. GetCLRValue maybe?

**Pull request merged**